operator/pkg/ciliumendpointslice/endpointslice.go

```diff
@@ -0,0 +1,329 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciliumendpointslice
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/operator/metrics"
+	operatorOption "github.com/cilium/cilium/operator/option"
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	capi_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	csv2 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2"
+	csv2a1 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/k8s/informer"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/time/rate"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+	// cesNamePrefix is the prefix name added for the CiliumEndpointSlice
+	// resource.
+	cesNamePrefix = "ces"
+
+	// defaultSyncBackOff is the default backoff period for cesSync calls.
+	defaultSyncBackOff = 1 * time.Second
+	// maxSyncBackOff is the max backoff period for cesSync calls.
+	maxSyncBackOff = 100 * time.Second
+	// maxRetries is the number of times a cesSync will be retried before it is
+	// dropped out of the queue.
+	maxRetries = 15
+	// CEPs are batched into a CES, based on its Identity
+	cesIdentityBasedSlicing = "cesSliceModeIdentity"
+	// default qps limit value for workqueues, this only for retries.
+	CESControllerWorkQueueQPSLimit = 10
+	// default burst limit value for workqueues.
+	CESControllerWorkQueueBurstLimit = 100
```

Comment:
I don't think we ever hit this, but isn't 100 too much? For whatever reason if we ever hit 100/s, apiserver would explode...I would set this to something like 20 which we know won't kill apiserver (at least in our tests).

Reply:
We had the same discussion earlier, client-go would limit the request going into k8s-apiserver.
As mentioned in the comment, burst limit is for workqueues.

Reply:
Can you add to the comment that this will be further limited by client qps?

Reply:
Existing comment is clearly states `burstLimit/qpsLimit` only for workqueues. It's implicit that client-go would limit that request going into k8s-apiserver.

Reply:
Then my question is, if client has a lower qps, what is the point of setting a higher number here?

Reply:
Workqueue QPS value is for the rate at which CES have to be processed irrespective of the the limits on the entire application QPS rate to talk with `k8s-apiserver`. Client-go QPS is configurable, customer can set the value at which they think it is comfortable for their cluster.